### DISCUSSION TITLE
Need help with this reconnecting issue

### DISCUSSION THREAD

**Discussion by NeLagina** [miguelgrinberg/python-socketio#1260] [Q&A] [answered]
**Describe the bug**
disconnecting when the function run too long

**To Reproduce**
1. Just get the packet and wait after it finished then its disconects and reconnects

**Expected behavior**
just staying connected

**Logs**

polling connection accepted with {'sid': '', 'upgrades': ['websocket'], 'pingInterval': 10000, 'pingTimeout': 5000, 'maxPayload': 1000000}
Engine.IO connection established
Sending packet MESSAGE data 0{}
Attempting WebSocket upgrade to wss://sockets.streamlabs.com/socket.io/?token=&transport=websocket&EIO=4
WebSocket upgrade was successful
Received packet MESSAGE data 0{"sid":"odGVo5njFBv05EZ7XxSR"}
Namespace / is connected
Prijunkta Prie StreamLabs Serverio
Received packet PING data
Sending packet PONG data
Received packet PING data
Sending packet PONG data
Received packet MESSAGE data 2["event",{"type":"donation","message":[{"name":"NeLagina","isTest":true,"formatted_amount":"€63.00","amount":63,"message":"This is a test donation for €63.00.","currency":"EUR","to":{"name":"NeLagina"},"from":"John","from_user_id":1,"_id":"9ae0477155d4ec1b8a04fa1def674eeb","priority":10}],"for":"streamlabs","event_id":"evt_65a09ead3ac38d2b00b876490f1789ea"}]
Received event "event" [/]
GAUTAS DONATIONAS ar kaskas tai lauk 10 sekundžiu
John Paukojo Šešiasdešimt Trys eurai
Žinute This is a test donation for €63.00.
Received packet PING data
Sending packet PONG data
packet queue is empty, aborting
Exiting write loop task
Server sent unexpected packet 8 data 0, aborting
Waiting for write loop task to end
Engine.IO connection dropped
Atsijunget Nuo StreamLabs Serverio
Connection failed, new attempt in 0.97 seconds
Exiting read loop task
Attempting polling connection to https://sockets.streamlabs.com/socket.io/?token=&transport=polling&EIO=4
Polling connection accepted with {'sid': '9Es4eQh6Fx3cuXubXxTj', 'upgrades': ['websocket'], 'pingInterval': 10000, 'pingTimeout': 5000, 'maxPayload': 1000000}
Engine.IO connection established
Sending packet MESSAGE data 0{}
Attempting WebSocket upgrade to wss://sockets.streamlabs.com/socket.io/?token=&transport=websocket&EIO=4
WebSocket upgrade was successful
Received packet MESSAGE data 0{"sid":"3t2Gi_WYL7ZwWmeZXxTl"}
Namespace / is connected
Prijunkta Prie StreamLabs Serverio
Reconnection successful

**Additional context**

**Comment by miguelgrinberg** [comment, marked as answer]
The issues board is for issues with this project, not for help with issues on your own application. Next time please use the discussions board to ask questions. I have moved your post this time.

The error is an unexpected packet from the server:

    packet queue is empty, aborting

This mean that the server stopped communicating with the client. The server indicated that it was going to send a `PING` packet every 10 seconds with the `'pingInterval': 10000`. If the client does not see PINGs at the declared interval it ends the connection assuming the server is unresponsive.

**Comment by NeLagina** [comment]
okay i will post it in discussions next time sorry for that , can you help me run like function separated that the socket ping and responds without waiting

**Comment by NeLagina** [comment]
nvm fixed

**Reply by azshimadathul** [reply]
how u fixed it?
